src/reader.rs

```diff
@@ -111,9 +111,9 @@ impl Reader {
         let memtable_iters = memtables
             .iter()
             .map(|t| t.range_ascending(range.clone()))
-            .collect();
+            .collect::<Vec<_>>();
```

Comment:
This seems weird to me. Is it possible to get the iterators from the memtables instead (the variable name even implies this). Right now it seems like we collect to vec then go back to iter on line 116

Reply:
yes, you're right, it makes more sense to remove this collect and directly take advantage of this `Iterator` after `map()`, let me fix this